src/grimoirelab/core/consumers/consumer.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) GrimoireLab Contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import json
+import logging
+import signal
+import time
+
+from collections import namedtuple
+from datetime import datetime, timedelta
+from typing import Iterable
+
+import redis
+
+
+ENTRIES_READ_COUNT = 10
+RECOVER_IDLE_TIME = 300000  # 5 minutes (in ms)
+STREAM_BLOCK_TIMEOUT = 60000  # 1 minute (in ms)
+
+
+Entry = namedtuple("Entry", ["message_id", "event"])
+
+
+class Consumer:
+    """Consumer base class to process events from a stream"""
+
+    def __init__(
+        self,
+        connection: redis.Redis,
+        stream_name: str,
+        consumer_group: str,
+        consumer_name: str,
+        stream_block_timeout: int = STREAM_BLOCK_TIMEOUT,
+        logging_level: str = "INFO",
+    ):
+        self.connection = connection
+        self.stream_name = stream_name
+        self.consumer_group = consumer_group
+        self.consumer_name = consumer_name
+        self.stream_block_timeout = stream_block_timeout
+        self.logging_level = logging_level
+        self.logger = self._create_logger()
+        self._stop = False
+        self._requested_stop_date: datetime | None = None
+
+    def start(self, burst: bool = False):
+        """Process events from the stream
+
+        This method first tries to recover entries from the stream that are
+        not yet processed by the consumer group in the last RECOVER_IDLE_TIME.
+
+        Once the entries are recovered, it starts to collect new entries
+        from the stream blocking for 'stream_block_timeout' if there aren't
+        new entries.
+        """
+        self.logger.info(f"Starting consumer '{self.consumer_name}' for '{self.stream_name}:{self.consumer_group}'")
+
+        self._setup_signal_handlers()
+        self._create_consumer_group()
+
+        while True:
+            recovered_entries = self.recover_stream_entries()
+            self.process_entries(recovered_entries)
+
+            new_entries = self.fetch_new_entries()
+            self.process_entries(new_entries)
+
+            if burst or self._stop:
+                self.logger.info(f"Consumer '{self.consumer_name}' stopped.")
+                break
+
+            time.sleep(1)
```

Comment:
This time should probably be a constant too. Not sure if 1 is too much. How often does rs pull for new jobs?